components/select/select-base.ts

```diff
@@ -70,6 +70,16 @@ export abstract class SuiSelectBase<T, U> implements AfterContentInit {
     }
 
     public set options(options:T[]) {
+        if (typeof (options) == "function") {
+            this.searchService.optionsLookup = options;
+            let subscription = this.dropdownService.isOpenChange.subscribe((value:boolean) => {
```

Comment:
You're probably asking about the `this.dropdownService.isOpenChange` subscription?

A dropdown is opened automatically when first receiving focus or clicking on a dropdown icon, and it display the no results message. I thought this was not a desired behaviour, so I changed it to do an empty query lookup. Maybe make this behaviour selectable by input parameter?